audit.go

```diff
@@ -505,6 +500,64 @@ func (c *AuditClient) getReply(seq uint32) (*syscall.NetlinkMessage, error) {
 	return &msg, nil
 }
 
+// unset our pid from the audit subsystem and close the socket.
+// This is a sort of isolated refactor, meant to deal with the deadlocks that can happen when we're not careful with blocking operations throughout a lot of this code.
+func (c *AuditClient) closeAndUnsetPid() error {
+	msg := syscall.NetlinkMessage{
+		Header: syscall.NlMsghdr{
+			Type:  AuditSet,
+			Flags: syscall.NLM_F_REQUEST,
+		},
+		Data: AuditStatus{
+			Mask: AuditStatusPID,
+			PID:  0,
+		}.toWireFormat(),
+	}
+
+	noParse := func(bytes []byte) ([]syscall.NetlinkMessage, error) {
+		return nil, nil
+	}
+
+	// If our request to unset the PID would block, then try to drain events from
+	// the netlink socket, resend, try again.
+	// In netlink, EAGAIN usually indicates our read buffer is full.
+	// The auditd code (which I'm using as a reference implementation) doesn't wait for a response when unsetting the audit pid.
+	maxLoop := 5
+	for i := 0; i < maxLoop; i++ {
+		_, err := c.Netlink.SendNoWait(msg)
+		// if we get an interrupt, retry the send
+		if err == nil {
+			return nil
+		} else if errors.Is(err, syscall.EINTR) {
```

Comment:
`(*NetlinkClient).SendNoWait` is a thin wrapper around `syscall.SendTo`. No functions in syscall ever wrap errors and given go1compat, this will not change (go1compat says that syscall is not covered, but this relates to changes forced by changes in the underlying OS and any change to types returned by syscalls would neither be likely from the Go team, nor quietly accepted by external reviewer).

tl;dr; No `errors.Is` is required here.

Reply:
so, even if `syscall.*` isn't going to be wrapping anything, I feel like it's safer to assume that `c.Netlink.*` might wrap something in the future.

Reply:
It shouldn't.

Reply:
My preferred approach would be to ensure that the behaviour is covered by tests, and then leave it without the extra weight. This would find regressions and ensure that bloat is not added to SendNoWait. I think the linter is wrong here. However, I'm not an owner, so it doesn't really matter.